Backend

chlouly opened the pull request:
Modified various parts of the backend, most importantly:

- Created the leave game endpoint that makes a player leave whatever game theyre in.
- Modified the joingame endpoint.

Commit:
Fixed the /joingame engdpoint

A few minor modifications were added to the three files listed above
to make the endpoint function correctly.

Commit:
Adding a leave game endpoint and updating input verrification.

mendez71 reviewed (approved):
Great Work! I love the changes to joing game, good work implementing the leave game endpoint

Commit:
Refactored endpoints
The joinrandom endpoint now functions a little differently.
When the gameNumber field is given "", the backend creates a new
game and returns the game number. When a number is given, it attempts
to join that game.

There is now a new type called ResCode in the ./backend/error.ts file,
it is an enum that contains a few custom error codes and with comments
showing their meanings.

All endpoints now only respond with JSON (when needed) and status codes,
no more empty messages.

Commit:
Added /logout endpoint
This endpoint saves the player's state to the RDS DB and removes them
from the online dictionary. The endpoint takes the id of the player in
the body.

RomanczuG reviewed (approved):
I love seeing that the new endpoint has been completed! I've never seen the end() function, but after reading about it, it seems pretty cool.
